app/geant-exporter/ImportProcessConverter.cc

```diff
@@ -499,5 +536,162 @@ void ImportProcessConverter::add_table(const G4PhysicsTable* g4table,
+                = element_physvec_map.find(elem_comp.element_id)->second;
+
+            for (int bin_idx : celeritas::range(energy_grid.size()))
+            {
+                const double energy_bin_value = energy_grid.at(bin_idx);
+
+                // Calculate cross-section
+                const double xs_per_atom
+                    = std::max(
+                          0.0,
+                          model.ComputeCrossSectionPerAtom(&g4_particle_def,
+                                                           &g4_element,
+                                                           energy_bin_value,
+                                                           particle_cutoff,
+                                                           energy_bin_value))
+                      / cm2;
+
+                physics_vector.y[bin_idx] = xs_per_atom;
+            }
+        }
+
+        // Avoid cross-section vectors starting or ending with zero values.
+        // Geant4 simply uses the next/previous bin value when the vector's
+        // front/back are zero. This feels like a trap.
+        for (auto& iter : element_physvec_map)
+        {
+            auto& physics_vector = iter.second;
+
+            if (physics_vector.y.front() == 0.0)
+            {
+                // Cross-section starts with zero, use next bin value
+                physics_vector.y.front() = physics_vector.y[1];
+            }
+
+            const unsigned int last_bin = physics_vector.y.size() - 1;
+            if (physics_vector.y.back() == 0.0)
+            {
+                // Cross-section ends with zero, use previous bin value
+                physics_vector.y.back() = physics_vector.y[last_bin - 1];
+            }
+        }
+
+        // Store element cross-section map for this material
+        element_selector.push_back(element_physvec_map);
+    }
+
+    return element_selector;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Set up \c ImportPhysicsVector type and energy grid, while leaving
+ * cross-section data empty. This method is used to initialize physics vectors
+ * in \c this->add_element_selector(...) .
+ *
+ * The energy grid is defined as in
+ * \c G4VEmModel::InitialiseElementSelectors(...) .
+ */
+ImportPhysicsVector
+ImportProcessConverter::initialize_element_selector_physics_vector(
+    G4VEmModel& model, const ImportMaterial& material)
+{
+    CELER_ASSERT(!material.elements.empty());
+
+    ImportPhysicsVector physics_vector;
+    physics_vector.vector_type = ImportPhysicsVectorType::log;
+
+    const double max_energy = model.HighEnergyLimit() / MeV;
+    const double min_energy
+        = material.pdg_cutoffs.find(process_.particle_pdg)->second.energy;
+
+    const int bins_per_decade
+        = G4EmParameters::Instance()->NumberOfBinsPerDecade();
+
+    const double inv_log_10_6 = 1.0 / (6.0 * std::log(10.0));
+    const int    num_energy_bins
+        = std::max(3,
```

Comment:
Okay, never mind for the comment on the maximum 3 - actually minimum 3 bins for the energy grid!